### PR TITLE
Feature/use userinfo instead of userdetails in idam

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1390

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by doncem — APPROVED** [review]

**Review by mdayican — APPROVED** [review]

File: src/test/resources/mappings/details-2caf7585-b23d-4ecc-80bf-50b0dbf83a62.json [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
   "name" : "details",
   "request" : {
-    "url" : "/details",
+    "url" : "/o/userinfo",
     "method" : "GET"
   },
   "response" : {
     "status" : 200,
-    "body" : "{\"forename\":\"Bulkscan\",\"surname\":\"Caseworker\",\"roles\":[\"caseworker-bulkscan\",\"caseworker\",\"caseworker-bulkscan-loa1\",\"caseworker-loa1\"],\"id\":640,\"defaultService\":\"CCD\",\"email\":\"bulkscan+ccd@gmail.com\"}",
+    "body" : "{\"forename\":\"Bulkscan\",\"surname\":\"Caseworker\",\"roles\":[\"caseworker-bulkscan\",\"caseworker\",\"caseworker-bulkscan-loa1\",\"caseworker-loa1\"],\"uid\":640,\"defaultService\":\"CCD\",\"email\":\"bulkscan+ccd@gmail.com\"}",
```
Comment (rkondratowicz) [inline src/test/resources/mappings/details-2caf7585-b23d-4ecc-80bf-50b0dbf83a62.json:10]: doesn't look like a valid /o/userinfo response

File: src/test/resources/mappings/details-2caf7585-b23d-4ecc-80bf-50b0dbf83a62.json [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
   "name" : "details",
   "request" : {
-    "url" : "/details",
+    "url" : "/o/userinfo",
     "method" : "GET"
   },
   "response" : {
     "status" : 200,
-    "body" : "{\"forename\":\"Bulkscan\",\"surname\":\"Caseworker\",\"roles\":[\"caseworker-bulkscan\",\"caseworker\",\"caseworker-bulkscan-loa1\",\"caseworker-loa1\"],\"id\":640,\"defaultService\":\"CCD\",\"email\":\"bulkscan+ccd@gmail.com\"}",
+    "body" : "{\"uid\":640,\"sub\":\"bulkscan+ccd@gmail.com\",\"roles\":[\"caseworker-bulkscan\",\"caseworker\",\"caseworker-bulkscan-loa1\",\"caseworker-loa1\"],\"name\":\"Bulkscan\",\"given_name\":\"Bulkscan\",\"family_name\":\"Caseworker\"}",
```
Comment (rkondratowicz) [inline src/test/resources/mappings/details-2caf7585-b23d-4ecc-80bf-50b0dbf83a62.json:10]: isn't uid a string?

File: src/test/resources/mappings/details-2caf7585-b23d-4ecc-80bf-50b0dbf83a62.json [hunk 3]
```diff
@@ -2,12 +2,12 @@
   "id" : "2caf7585-b23d-4ecc-80bf-50b0dbf83a62",
```
Comment (rkondratowicz) [inline src/test/resources/mappings/details-2caf7585-b23d-4ecc-80bf-50b0dbf83a62.json:2]: I think file should be renamed (name matched url before)

**Review by rkondratowicz — CHANGES_REQUESTED** [review]
fix wiremock response

**Review by rkondratowicz — APPROVED** [review]

**Pull request merged**